dspy/__init__.py

```diff
@@ -64,6 +66,13 @@
 configure = settings.configure
 context = settings.context
 
+CACHE_DIR = os.environ.get("DSPY_CACHEDIR") or os.path.join(Path.home(), ".dspy_cache")
```

Comment:
Note to self: .dspy_cache is already used for litellm, this PR should be swapped (it uses .dspy_litellm_cache for the existing cache but that cache doesn't exist at that new path)